### DISCUSSION TITLE
jsonb_matches_schema showing up not existing even when extension enabeled.

### DISCUSSION THREAD

**Discussion by countzen2000** [supabase/supabase#22032] [Questions] [answered]
I have `pg_jsonschema` extension enabaled for the database, this adds (amongst other things)`jsonb_matches_schema` and `json_matches_schema` as function that are used for validating schemas.

When I call the edge function on supabase writing to a jsonb column, I get the error:
`{"error":"function jsonb_matches_schema(json, jsonb) does not exist"}`

If I replicate the database locally, with supabase db reset and dump etc. the edge function works as per normal.

Has anyone encountered this?
<img width="596" alt="Screenshot 2024-03-14 at 12 08 01 AM" src="https://github.com/supabase/supabase/assets/202204/4814e722-f525-40a3-8741-d76f5abbf134">

<img width="646" alt="Screenshot 2024-03-12 at 10 07 54 PM" src="https://github.com/supabase/supabase/assets/202204/9d51eac2-4a60-4fc3-b35e-e73041579aff">

**Comment by countzen2000** [comment, marked as answer]
Ok this was very bad/dumb on Supabase. I wrote a new script to take off all existing constraints that uses `jsonb_matches_schema`, then I turned off the extension and turned it back on again in extension UI window in supabse. Then ran another script that added all the constraints I took off. 

And now it works.

What in the world, I have never encountered this in a normal postgres deployment.

**Reply by kierancrown** [reply]
Did you just re-enable? I'm having this issue when both installing via SQL and via the dashboard and cannot get it to work
